flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/harness/CountTriggerWithCleanupStateHarnessTest.scala

```diff
@@ -0,0 +1,305 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.runtime.harness
+
+import com.google.common.collect.Lists
+import org.apache.flink.api.common.time.Time
+import org.apache.flink.streaming.api.windowing.triggers.TriggerResult
+import org.apache.flink.streaming.api.windowing.windows.TimeWindow
+import org.apache.flink.streaming.runtime.operators.windowing.TriggerTestHarness
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord
+import org.apache.flink.table.api.StreamQueryConfig
+import org.apache.flink.table.runtime.triggers.CountTriggerWithCleanupState
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class CountTriggerWithCleanupStateHarnessTest {
+  protected var queryConfig =
+    new StreamQueryConfig().withIdleStateRetentionTime(Time.seconds(2), Time.seconds(3))
+
+  @Test
+  def testFiringAndFireingWithPurging(): Unit = {
+    val testHarness = new TriggerTestHarness[Any, TimeWindow](
+      CountTriggerWithCleanupState.of[TimeWindow](queryConfig, 10), new TimeWindow.Serializer)
+
+    // try to trigger onProcessingTime method via 1, but there is non timer is triggered
+    assertEquals(0, testHarness.advanceProcessingTime(1).size())
+
+    // register cleanup timer with 3001
+    assertEquals(
+      TriggerResult.CONTINUE,
+      testHarness.processElement(new StreamRecord[Any](1), new TimeWindow(0, 9)))
```

Comment:
use same `StreamRecord` and `GlobalWindow` object to keep the code more concise?